### PR TITLE
JDK-8324637: [aix] Implement support for reporting swap space in jdk.management

### PR DESCRIPTION
**Pull request by MBaesken**
The get_total_or_available_swap_space_size coding misses AIX support, we only return 0. This should be enhanced.
The perfstat API can be used, see https://www.ibm.com/docs/pt/aix/7.2?topic=interfaces-perfstat-memory-total-interface .

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324637](https://bugs.openjdk.org/browse/JDK-8324637): [aix] Implement support for reporting swap space in jdk.management (**Bug** - P4)

### Reviewers
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17569/head:pull/17569` \
`$ git checkout pull/17569`

Update a local copy of the PR: \
`$ git checkout pull/17569` \
`$ git pull https://git.openjdk.org/jdk.git pull/17569/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17569`

View PR using the GUI difftool: \
`$ git pr show -t 17569`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17569.diff">https://git.openjdk.org/jdk/pull/17569.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17569#issuecomment-1910116197)

### CODE REVIEW

**Review by kevinjwalls — APPROVED** [review]
I can't try this and don't use AIX, but it looks good.
It follows the same pattern as the other AIX cases in the file.

Although the others (e.g. line 200) don't throw_internal_error if the call returns -1, they just return -1.  You might want to do that for consistency of behaviour.

File: src/jdk.management/unix/native/libmanagement_ext/OperatingSystemImpl.c [hunk 1]
```diff
@@ -105,6 +105,12 @@ static jlong get_total_or_available_swap_space_size(JNIEnv* env, jboolean availa
         throw_internal_error(env, "sysctlbyname failed");
     }
     return available ? (jlong)vmusage.xsu_avail : (jlong)vmusage.xsu_total;
+#elif defined(_AIX)
+    perfstat_memory_total_t memory_info;
+    if (perfstat_memory_total(NULL, &memory_info, sizeof(perfstat_memory_total_t), 1) == -1) {
+        throw_internal_error(env, "perfstat_memory_total failed");
+    }
+    return available ? (jlong)(memory_info.pgsp_free * 4L * 1024L) : (jlong)(memory_info.pgsp_total * 4L * 1024L);
```
Comment (tstuefe) [inline src/jdk.management/unix/native/libmanagement_ext/OperatingSystemImpl.c:113]: Do we need the cast? perfstat_memory_total_t members are all 64-bit, no?

Also, can we shorten this to:

```
return (available ? memory_info.pgsp_free : memory_info.pgsp_total) * 4096;
```

**Review by tstuefe — APPROVED** [review]
Small nit, otherwise good.

**Comment by MBaesken** [comment]
> I can't try this and don't use AIX, but it looks good. It follows the same pattern as the other AIX cases in the file.
> 
> Although the others (e.g. line 200) don't throw_internal_error if the call returns -1, they just return -1. You might want to do that for consistency of behaviour.

Hi I wanted to be consistent to the other OS in get_total_or_available_swap_space_size, that#s why the throw_internal_error .

File: src/jdk.management/unix/native/libmanagement_ext/OperatingSystemImpl.c (hunk 1 above)
Comment (MBaesken) [reply src/jdk.management/unix/native/libmanagement_ext/OperatingSystemImpl.c:113]: > Do we need the cast? perfstat_memory_total_t members are all 64-bit, no?
> 
> Also, can we shorten this to:
> 
> ```
> return (available ? memory_info.pgsp_free : memory_info.pgsp_total) * 4096;
> ```

Hi Thomas, I see no types defined here https://www.ibm.com/docs/pt/aix/7.2?topic=interfaces-perfstat-memory-total-interface but most likely you are correct and they are 64bit.

**Comment by kevinjwalls** [comment]
> > I can't try this and don't use AIX, but it looks good. It follows the same pattern as the other AIX cases in the file.
> > Although the others (e.g. line 200) don't throw_internal_error if the call returns -1, they just return -1. You might want to do that for consistency of behaviour.
> 
> Hi I wanted to be consistent to the other OS in get_total_or_available_swap_space_size, that#s why the throw_internal_error .

Ok yes, you can choose who to be consistent with... 8-)
